DevIncubator_Autopark/AllCollections/MyStack.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace DevIncubator_Autopark.AllCollections
+{
+    public class MyStack<T> : IEnumerable<T>
+    {
+        public int Count => GetCount();
+
+        private T[] _stack;
+        private int _headItemIndex;
+
+        public MyStack()
+        {
+            int defaultCapacity = 10;
+            _stack = new T[defaultCapacity];
+
+            _headItemIndex = -1;
+        }
+
+        public MyStack(IEnumerable<T> stack)
+        {
+            if (stack == null)
+            {
+                throw new ArgumentNullException(nameof(stack), "The transmitted data cannot be null");
+            }
+
+            foreach (var item in stack)
+            {
+                Push(item);
+            }
```

Comment:
When you push() the first element, his index is 1 `46. _stack[++_headItemIndex] = item;` But his index should be 0.

Reply:
fixed